gocourse04/zoo/zoo.go

```diff
@@ -0,0 +1,93 @@
+package zoo
+
+import "fmt"
+
+type Zoo struct {
+	Areas      Areas
+	SetOfNames map[string]struct{} // smthiong like bloom filter
+}
+
+func NewZoo() *Zoo {
+	return &Zoo{
+		Areas:      make(map[AnimalSpecies]Area),
+		SetOfNames: make(map[string]struct{}),
+	}
+}
+
+func (z *Zoo) AddAnimal(animal *Animal) (bool, error) {
+	if _, exists := z.SetOfNames[animal.Name]; exists {
+		return false, fmt.Errorf("Animal %s already exists", animal.Name)
```

Comment:
https://go.dev/wiki/CodeReviewComments#error-strings

> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context

```suggestion
		return false, fmt.Errorf("animal %s already exists", animal.Name)
```